File: tests/java/src/main/java/com/escape/tests/cases/Case135ThreadHandoffLate04.java
package com.escape.tests.cases;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Supplier;

/** Task case 135: thread_handoff_late_04 false-positive/false-negative stress pattern. */
public class Case135ThreadHandoffLate04 {
    private static final Map<String, Map<String, String>> RETAINED_CACHE = new HashMap<>();
    private static final List<Map<String, String>> RETAINED_AUDIT = new ArrayList<>();
    private static final List<Supplier<String>> RETAINED_HANDLERS = new ArrayList<>();

    public static String execute(String input) {
        String taskName = "thread_handoff_late_04";
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", taskName);
        payload.put("entity", "stress");
        payload.put("stage", "evaluation");
        payload.put("input", raw);
        payload.put("checksum", taskName + ":" + raw.length());
        // ESCAPE: payload captured by background thread.
        new Thread(() -> RETAINED_AUDIT.add(payload)).start();
        return "ok";
    }
}

File: tests/go/cases/case_135_thread_handoff_late_04.go
package escape_tests

var retainedCase135 = []map[string]string{}

func Case135ThreadHandoffLate04(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "thread_handoff_late_04",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "thread_handoff_late_04:" + raw,
	}
	// ESCAPE: goroutine captures payload and appends to retained sink.
	go func(p map[string]string) {
		retainedCase135 = append(retainedCase135, p)
	}(payload)
	return "ok"
}
